src/jailer/src/env.rs

```diff
@@ -609,6 +609,80 @@ mod tests {
                 metadata.permissions().mode(),
                 libc::S_IFCHR | libc::S_IRUSR | libc::S_IWUSR
             );
+
+            // Trying to create again the same device node is not allowed.
+            assert_eq!(
+                format!("{}", env.mknod_and_own_dev(dev, major, minor).unwrap_err()),
+                format!(
+                    "Failed to create {}\u{0} via mknod inside the jail: File exists (os error 17)",
+                    dev_str
+                )
+            );
+            // Remove the device node.
+            fs::remove_file(dev_str).expect("Could not remove file.");
```

Comment:
Could we use`TempFile::with_prefix` instead of `/dev/net/tun-test` etc.? This way we make sure that the device files are deleted even if this test function quits before reaching `remove_file`.

Reply:
The files are created inside `mknod_and_own_dev()`. This function takes a `&'static [u8]` as parameter, I don't think `Tempfile` could help here.

I added an extra check at the beginning to be sure the file doesn't exist before attempting to create it, should be safer now :-??.